tests/e2e/pageobjects/git-providers/OauthPage.ts

```diff
@@ -17,140 +17,186 @@ import { TimeoutConstants } from '../../constants/TimeoutConstants';
-        Logger.debug('OauthPage.waitOauthPage');
+    await this.driverHelper.waitDisappearance(this.loginForm);
+  }
 
-        await this.driverHelper.waitVisibility(this.approveButton);
-    }
+  async waitOauthPage(): Promise<void> {
+    Logger.debug('OauthPage.waitOauthPage');
 
-    async clickOnApproveButton(): Promise<void> {
-        Logger.debug('OauthPage.clickOnApproveButton');
+    await this.driverHelper.waitVisibility(this.approveButton);
+  }
 
-        await this.driverHelper.waitAndClick(this.approveButton);
-    }
+  async clickOnApproveButton(): Promise<void> {
+    Logger.debug('OauthPage.clickOnApproveButton');
 
-    async clickOnDenyAccessButton(): Promise<void> {
-        Logger.debug('OauthPage.clickOnDenyAccessButton');
+    await this.driverHelper.waitAndClick(this.approveButton);
+  }
 
-        await this.driverHelper.waitAndClick(this.denyAccessButton);
-    }
+  async clickOnDenyAccessButton(): Promise<void> {
+    Logger.debug('OauthPage.clickOnDenyAccessButton');
 
-    async waitDisappearanceOauthPage(): Promise<void> {
-        Logger.debug('OauthPage.waitDisappearanceOauthPage');
+    await this.driverHelper.waitAndClick(this.denyAccessButton);
+  }
 
-        await this.driverHelper.waitDisappearance(this.approveButton);
-    }
+  async waitDisappearanceOauthPage(): Promise<void> {
+    Logger.debug('OauthPage.waitDisappearanceOauthPage');
 
-    async login(): Promise<void> {
-        Logger.debug('OauthPage.login');
+    await this.driverHelper.waitDisappearance(this.approveButton);
+  }
 
-        await this.waitLoginPage();
-        await this.enterUserName(TestConstants.TS_SELENIUM_GIT_PROVIDER_USERNAME);
-        await this.enterPassword(TestConstants.TS_SELENIUM_GIT_PROVIDER_PASSWORD);
-        await this.clickOnLoginButton();
-        await this.waitClosingLoginPage();
-    }
+  async login(): Promise<void> {
+    Logger.debug('OauthPage.login');
 
-    async confirmAccess(): Promise<void> {
-        Logger.debug('OauthPage.confirmAccess');
-
-        try {
-            await this.clickOnApproveButton();
-            await this.waitDisappearanceOauthPage();
-        } catch (e) {
-            Logger.debug('OauthPage.confirmAccess - access was not confirmed, retrying to click conformation button');
-            // workaround for github oauth conformation page (bot security)
-            await this.driverHelper.getAction()
-                .move({
-                    origin: await this.driverHelper.waitPresence(this.approveButton)
-                })
-                .click()
-                .perform();
-            await this.waitDisappearanceOauthPage();
-        }
-    }
+    await this.waitLoginPage();
+    await this.enterUserName(TestConstants.TS_SELENIUM_GIT_PROVIDER_USERNAME);
+    await this.enterPassword(TestConstants.TS_SELENIUM_GIT_PROVIDER_PASSWORD);
+    await this.clickOnLoginButton();
+    await this.waitClosingLoginPage();
+    await this.waitDisappearanceTwoFactorInfoBox();
```

Comment:
As I can see, 2FA is applied to any scope, but as I remember we don't have it, for example here _https://bitbucket-bitbucket.apps.ds-airgap-v12.crw-qe.com_ or here _https://gitlab.cee.redhat.com_ for factory tests. Did you try to run some test without it? Maybe it is better to add some env variable, like: 
`TestConstants.TS_SELENIUM_IS_2FA_ENABLED === 'true'` ?

Reply:
I could but the check will fall through when there's no 2FA to wait for so I think another TestConstant isn't required.